From 1ed7236dca138a5eb1eed2f0e5b27f15fb1535d2 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:29:59 -0400
Subject: [alsa] hda: Use runtime helper to check active state

Message-id: <140972823573-262-patchset-jkysela@redhat.com>
Patchwork-id: 90962
O-Subject: [RHEL 7.1 PATCH 261/442] ALSA: hda - Use runtime helper to check active state.
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

>From azx_interrupt, use the helper to check if the device is active
instead of checking the state.  This will do the right thing if
runtime pm is disabled in addition to if the device is suspended.

Signed-off-by: Dylan Reid <dgreid@chromium.org>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 7b0a48f3402a43bd8bb85bfda4f72e0a34634d74)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/hda_controller.c b/sound/pci/hda/hda_controller.c
index 3e005e5..8197afc 100644
--- a/sound/pci/hda/hda_controller.c
+++ b/sound/pci/hda/hda_controller.c
@@ -1759,7 +1759,7 @@ irqreturn_t azx_interrupt(int irq, void *dev_id)
 
 #ifdef CONFIG_PM_RUNTIME
  if (chip->driver_caps & AZX_DCAPS_PM_RUNTIME)
-  if (chip->card->dev->power.runtime_status != RPM_ACTIVE)
+  if (!pm_runtime_active(chip->card->dev))
    return IRQ_NONE;
 #endif
 
-- 
1.7.1